### DISCUSSION TITLE
Automate visual mode

### DISCUSSION THREAD

**Discussion by pipothebit** [radareorg/radare2#20227] [Q&A] [answered]
Is there any way to "automate" the visual mode? I'm talking about using pipes to execute commands like go to, cursor up, down, etc. Something like sending keys from pipes to radar visual mode.

**Comment by trufae** [comment, marked as answer]
Its easy to do with vte, but if you are not in the mood to create a new terminal emulator you can use tmux that allows to interact with any terminal app from another session by sending keys or dumping the contents of it.

**Comment by pipothebit** [comment]
Perfect, it works.
I need to measure and save to later analyze the execution times of pieces of code. I know that radare2 is not designed for this but it does the job. I wanted to save the time as well but I couldn't redirect the result of ?t xxx to a file.

I have changed that on my working copy of r2 and wanted to discuss why this is currently the case? Would it be useful to change it?
```
diff --git a/libr/core/cmd_help.c b/libr/core/cmd_help.c
index bc5719d0d..f4df83117 100644
--- a/libr/core/cmd_help.c
+++ b/libr/core/cmd_help.c
@@ -586,7 +586,7 @@ static int cmd_help(void *data, const char *input) {
                                r_core_cmd (core, input + 1, 0);
                                r_prof_end (&prof);
                                r_core_return_value (core, (ut64)(int)prof.result);
-                               eprintf ("%lf\n", prof.result);
+                               printf ("%lf\n", prof.result);
                                break;
                        }
                default:
```

Now I can do this:
```
?t aecu 0x00002054 >> profiler.log
```

**Comment by pipothebit** [comment]
I answer to myself.
No need, r2 is capable of redirecting stderr. Problem of being a newbie and not reading the documentation. Sorry about the noise, delete the question?
```
?t aecu 0x00002054 2>> profiler.log
```
https://r2wiki.readthedocs.io/en/latest/options/redirection/#redirection
